### PR TITLE
[CLOUDTRUST-2792] Upgrade to Keycloak 14.0.0

### PR DESCRIPTION
**Pull request by fperot74**

### CODE REVIEW

File: keycloak-export/src/main/java/io/cloudtrust/keycloak/export/SingleFileImportProvider.java [hunk 1]
```diff
@@ -49,12 +47,10 @@ public void importModel(KeycloakSessionFactory factory, Strategy strategy) throw
 
             if (masterRealm != null) {
                 // If master was imported, we may need to re-create realm management clients
-                for (RealmModel realm : session.realms().getRealms()) {
-                    if (realm.getMasterAdminClient() == null) {
-                        logger.infof("Re-created management client in master realm for realm '%s'", realm.getName());
-                        new RealmManager(session).setupMasterAdminManagement(realm);
-                    }
-                }
+                session.realms().getRealmsStream().filter(r -> r.getMasterAdminClient()==null).forEach(realm -> {
```
Comment (lagess) [inline keycloak-export/src/main/java/io/cloudtrust/keycloak/export/SingleFileImportProvider.java]: small formatting issues ( == )

File: keycloak-export/src/main/resources/theme/ctexport/admin/resources/js/services.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
```
Comment (lagess) [inline keycloak-export/src/main/resources/theme/ctexport/admin/resources/js/services.js:2]: What kind of customisations do we have in this file? Don't we want to make them explicit? (e.g. with from BEGIN CUSTO.... END CUSTO tags)

File: keycloak-export/src/main/resources/theme/ctexport/admin/resources/js/controllers/realm.js [hunk 3]
```diff
@@ -83,7 +83,7 @@ function getAccessObject(Auth, Current) {
 }
```
Comment (lagess) [inline keycloak-export/src/main/resources/theme/ctexport/admin/resources/js/controllers/realm.js:83]: What kind of customisations do we have in this file? Don't we want to make them explicit? (e.g. with from BEGIN CUSTO.... END CUSTO tags)

File: keycloak-export/src/main/java/io/cloudtrust/keycloak/export/ExportResourceProvider.java [hunk 4]
```diff
@@ -109,19 +109,17 @@ public RealmRepresentation exportRealm(@Context final HttpHeaders headers, @Cont
...
 
-        for (UserModel user : session.users().getUsers(realm, true)) {
+        session.users().getUsersStream(realm, true).forEach(user -> {
             UserRepresentation userRep = userRepMap.get(user.getId());
             if (userRep != null) {
                 // Credentials
-                List<CredentialModel> creds = session.userCredentialManager().getStoredCredentials(realm, user);
-                List<CredentialRepresentation> credReps = creds.stream().map(this::exportCredential).collect(Collectors.toList());
+                Stream<CredentialModel> creds = session.userCredentialManager().getStoredCredentialsStream(realm, user);
+                List<CredentialRepresentation> credReps = creds.map(this::exportCredential).collect(Collectors.toList());
```
Comment (lagess) [inline keycloak-export/src/main/java/io/cloudtrust/keycloak/export/ExportResourceProvider.java]: we could chain this line and the above one, no?

**Review by lagess — APPROVED** [review]

**Review by harture — APPROVED** [review]

**Pull request merged**